spec/support/cuprite/custom_drivers.rb

```diff
@@ -0,0 +1,38 @@
+module Cuprite::CustomDrivers
+  DOMAIN_RESTRICTED_CUPRITE = :domain_restricted_cuprite
+  UNRESTRICTED_CUPRITE = :unrestricted_cuprite
+
+  class << self
+    def register_with_capybara
+      register_driver(
+        DOMAIN_RESTRICTED_CUPRITE,
+        -> { { url_whitelist: %r{\Ahttp://localhost:#{Capybara.server_port}/} } },
```

Comment:
This needs to be wrapped in a proc to delay its execution (I guess until we are within the `Capybara.register_driver` block), because otherwise `Capybara.server_port` is not yet defined, and so basically nothing gets allowlisted and pretty much every spec fails.